snapshot-labs/sx-ui
# feat: add Highlight votes support

### Summary

This PR adds support for Highlight votes and incorporates them into UI.
Votes are accounted in:
- Vote count on spaces and proposals
- Proposal scores
- Votes pages

Limitations:
- As Votes are coming from separate APIs it's hard to do proper sorting and filtering for Votes page with lazy loading. Right now we will fetch 1000 votes from Highlight initially and then on-demand will feed them into Votes requests to be properly mixed in with regular results. Long term solution would be to have single API to handle all of those.
- Currently there is single API that will be enabled for all EVM networks.
- Open transaction link for Vote is not connected to anything.

Closes: https://github.com/snapshot-labs/pitches/issues/54

### How to test

1. Start highlight on this PR: https://github.com/snapshot-labs/highlight/pull/9.
2. Run highlight.js tests on this PR: https://github.com/snapshot-labs/highlight.js/pull/3 (this inserts test Vote).
3. Set `VITE_HIGHLIGHT_URL=http://localhost:3000`. 
4. Interact with this space http://localhost:8080/#/matic:0xa98d476118af1a2c9eab803db9688e33cf68435f
5. Space vote count is higher with highlight enabled.
6. Check this proposal: http://localhost:8080/#/matic:0xa98d476118af1a2c9eab803db9688e33cf68435f
7. It shows additional against vote in results.
8. Votes modal and Votes tab works as expected (other than Open transaction link).

### Screenshots

<img width="792" alt="image" src="https://github.com/snapshot-labs/sx-ui/assets/1968722/1d14817b-0178-4bee-9b61-8e1eff18d44b">
<img width="1019" alt="image" src="https://github.com/snapshot-labs/sx-ui/assets/1968722/cca6d246-d824-4c14-bd08-aac916d2c6f7">
<img width="1015" alt="image" src="https://github.com/snapshot-labs/sx-ui/assets/1968722/cede15a3-dcc9-4b3f-a329-06732c8f4d31">
